templates/workloads/partials/cf/service-base-properties.yml

```diff
@@ -2,7 +2,9 @@ Cluster:
   Fn::ImportValue:
     !Sub '${AppName}-${EnvName}-ClusterId'
 TaskDefinition: !Ref TaskDefinition
-{{- if .Autoscaling }}
+{{- if .DesiredCountOnSpot}}
```

Comment:
Maybe we don't need this condition?

Reply:
We need this since if desiredCount is specified via the `spot` field, then we don't want the Autoscaling lambda to be part of the stack.

Reply:
I might misunderstand something but I still feel this is a bit redundant since we already handle all the logic for desired count in https://github.com/aws/copilot-cli/pull/2188/files#diff-ac8fb9fb1f51e4365619b020fc6ccd1c32aab7ba83a8fce9e6a6e9ea7a341170R97 . The logic i think should be: if there's any autoscaling then we use the lambda value, if not, we use the `TaskCount`.